pyogrio/_io.pyx

```diff
@@ -1126,6 +1126,14 @@ def ogr_read(
         if sql is None:
             # layer defaults to index 0
             if layer is None:
+                layers = ogr_list_layers_dataset(ogr_dataset)
+                if len(layers) > 1:
+                    warnings.warn(
+                        f"More than one layer found in '{path}', layer {layers[0][0]} "
+                        f"is read, available layers: {layers[:, 0].tolist()}. Specify "
+                        "layer parameter to avoid this warning.",
```

Comment:
This can be quite verbose. Can we try to use some shorter version? I don't think we need to print `path` (user knows what they're reading). Same for the last sentence. We could say something like "Unspecified layer. Using  {layers[0][0]}, available layers: {layers[:, 0].tolist()}". It tells the same but is much shorter and it implies that specifying the layer silences the warning.

Reply:
No problem to try making the warning shorter, but I'm not sure about removing the path. Because there are sometimes different code paths in pyogrio that lead to the warning being triggered, the stacklevel isn't always good enough to lead to the line number in the user's script being shown. In a larger script where many files are being read I can imagine that it can be a hunt to find the relevant file.

Maybe just print the file name? Something like: "Unspecified layer, using {layers[0][0]}. Available layers in file {filename}: {{layers[:, 0].tolist()}"

Reply:
The warning is a bit shorter now (based on suggestion of Brendan).